src/components/ha-picture-upload.ts

```diff
@@ -141,16 +175,58 @@ export class HaPictureUpload extends LitElement {
     }
   }
 
+  private _chooseMedia(): void {
+    showMediaBrowserDialog(this, {
+      action: "pick",
+      entityId: "browser",
+      navigateIds: [
+        { media_content_id: undefined, media_content_type: undefined },
+        {
+          media_content_id: "media-source://image_upload",
+          media_content_type: "app",
+        },
+      ],
+      minimumNavigateLevel: 2,
+      mediaPickedCallback: async (pickedMedia: MediaPickedEvent) => {
+        const id = pickedMedia.item.media_content_id;
+        const stringToRemove = "media-source://image_upload/";
+        if (id.startsWith(stringToRemove)) {
+          const mediaId = id.substr(stringToRemove.length);
```

Comment:
`substr` is depricated: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr

Please use `substring`